src/controllers/dynakube/dynakube_controller.go

```diff
@@ -311,85 +317,23 @@ func updatePhaseIfChanged(instance *dynatracev1beta1.DynaKube, newPhase dynatrac
 
 func (controller *DynakubeController) removeOneAgentDaemonSet(dkState *status.DynakubeState) {
 	ds := appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: dkState.Instance.OneAgentDaemonsetName(), Namespace: dkState.Instance.Namespace}}
-	if err := controller.ensureDeleted(&ds); dkState.Error(err) {
+	if err := kubeobjects.EnsureDeleted(context.TODO(), controller.client, &ds); dkState.Error(err) {
 		return
 	}
 }
 
-func (controller *DynakubeController) ensureDeleted(obj client.Object) error {
-	if err := controller.client.Delete(context.TODO(), obj); err != nil && !k8serrors.IsNotFound(err) {
-		return err
-	}
-	return nil
-}
-
-func (controller *DynakubeController) reconcileActiveGate(ctx context.Context, dynakubeState *status.DynakubeState, dtc dtclient.Client) bool {
-	if !controller.reconcileActiveGateProxySecret(ctx, dynakubeState) {
+func (controller *DynakubeController) reconcileActiveGate(ctx context.Context, dynakubeState *status.DynakubeState, dtc dtclient.Client) (updated bool) {
+	upd, err := activegate.NewReconciler(ctx, controller.client, controller.apiReader, controller.scheme, dynakubeState.Instance).Reconcile()
```

Comment:
introduces a dependency on an activegate reconciler instance that cannot be mocked / faked for testing